src/afrolabs/components/kafka/msk.clj

```diff
@@ -0,0 +1,96 @@
+(ns afrolabs.components.kafka.msk
+  (:require
+   [afrolabs.components.kafka :as -kafka]
+   [taoensso.timbre :as log]))
+
+
+(-kafka/defstrategy AwsMskScram
+  ;; "Sets the required and recommended config to connect to a kafka cluster in confluent cloud.
+  ;; Based on: https://docs.confluent.io/cloud/current/client-apps/config-client.html#java-client"
+  [& {:aws.msk/keys [username password]}]
+
+  (log/info (str "Connection to kafka configured with AwsMskScram strategy..."
+                 (when-not (and username password)
+                   " (don't have username & password.)")))
+
+  (letfn [(merge-common
+            [cfg]
+            (cond-> cfg
+              true
+              (merge {"client.dns.lookup"        "use_all_dns_ips"
+                      "reconnect.backoff.max.ms" "10000"
+                      "request.timeout.ms"       "30000"})
+
+              (and username (seq username)
+                   password (seq password))
```

Comment:
`seq` works properly in the presence of `nil` so this can just be `(and (seq username) (seq password))`